### PR TITLE
feat: Adding av commit amend cmd

### PR DESCRIPTION
**Pull request by ohcnivek**
### test: 
Screen Shot 2023-05-26 at 1 30 47 AM

### CODE REVIEW

File: cmd/av/commit_amend.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+ func commitAmend(repo *git.Repo, currentBranchName string, flags struct{Message string; NoEdit bool}) error {
+	// Argument validation
+	if flags.NoEdit && flags.Message != "" {
+		return errors.New("cannot use --no-edit & amend the commit message.")
+	}
+
+	
+	commitArgs := []string{"commit"}
+	commitArgs = append(commitArgs, "--amend")
```
Comment (travigd) [inline cmd/av/commit_amend.go:61]:
```suggestion
	commitArgs := []string{"commit", "--amend"}
```

File: cmd/av/commit_amend.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		return nil
+	},
+}
+
+func init() {
+	commitAmendCmd.Flags().StringVarP(&commitAmendFlags.Message, "message", "m", "", "the commit message")
+	commitAmendCmd.Flags().BoolVarP(&commitAmendFlags.NoEdit, "no-edit", "a", false, "amend a commit without changing its commit message")
+}
+
+ func commitAmend(repo *git.Repo, currentBranchName string, flags struct{Message string; NoEdit bool}) error {
```
Comment (travigd) [inline cmd/av/commit_amend.go:53]: Just put this inline with `RunE`.

File: cmd/av/commit_amend.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	// Argument validation
+	if flags.NoEdit && flags.Message != "" {
+		return errors.New("cannot use --no-edit & amend the commit message.")
```
Comment (travigd) [inline cmd/av/commit_amend.go:56]: Use `cobra.Cmd.MarkFlagsMutuallyExclusive` instead: https://github.com/aviator-co/av/blob/e7e8cfe9cc7d9bbdf3a5c0c5da1c240928f837ce/cmd/av/stack_sync.go#L308

File: cmd/av/commit_amend.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+			return err
+		}
+
+		return nil
+	},
+}
+
+func init() {
+	commitAmendCmd.Flags().StringVarP(&commitAmendFlags.Message, "message", "m", "", "the commit message")
+	commitAmendCmd.Flags().BoolVarP(&commitAmendFlags.NoEdit, "no-edit", "a", false, "amend a commit without changing its commit message")
```
Comment (travigd) [inline cmd/av/commit_amend.go:50]: Let's not add shorthand versions that don't exist in Git. You can add a non-shorthand flag with just `BoolVar`.

File: cmd/av/commit_amend.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+			commitArgs = append(commitArgs, "--message", commitAmendFlags.Message)
+		}
+
+		if _, err := repo.Run(&git.RunOpts{
+			Args:        commitArgs,
+			ExitError:   true,
+			Interactive: true,
+		}); err != nil {
+			_, _ = fmt.Fprint(os.Stderr,
+				"\n", colors.Failure("Failed to create commit."), "\n",
```
Comment (draftcode) [inline cmd/av/commit_amend.go:54]: Failed to amend?

**Review by draftcode — APPROVED** [review]

**Pull request closed without merging**